lib/Service/BanService.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * SPDX-FileCopyrightText: 2023 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Talk\Service;
+
+use DateTime;
+use OCA\Talk\Model\Ban;
+use OCA\Talk\Model\BanMapper;
+use OCP\AppFramework\Db\DoesNotExistException;
+use OCP\AppFramework\Utility\ITimeFactory;
+
+class BanService {
+	
+	public function __construct(
+		protected BanMapper $banMapper,
+		protected ITimeFactory $timeFactory,
+	) {
+	}
+
+	/**
+	 * Validate the ban data.
+	 */
+	private function validateBanData(string $actorId, string $actorType, int $roomId, string $bannedId, string $bannedType, ?DateTime $bannedTime, ?string $internalNote): void {
+		if (empty($actorId) || empty($actorType) || empty($roomId) || empty($bannedId) || empty($bannedType)) {
+			throw new \InvalidArgumentException("Invalid ban data provided.");
+		}
+
+		if ($bannedTime !== null && !$bannedTime instanceof DateTime) {
+			throw new \InvalidArgumentException("Invalid date format for bannedTime.");
+		}
```

Comment:
Should expend this and have an exception for each of them with their own "key" (not "Invalid ban data provided", but `invalid_banned`, `invalid_note`, …) (where it makes sense, e.g. roomId can never be invalid and neither can "actor" as it's not provided but generated inside our code)